felix/proto/felixbackend.proto

```diff
@@ -566,16 +579,25 @@ message WireguardEndpointUpdate {
   // The name of the wireguard host.
   string hostname = 1;
 
-  // The public key for this endpoint.
+  // The public key for IPv4 wireguard on this endpoint.
   string public_key = 2;
 
-  // The IP address of the wireguard interface.
+  // The IP address of the IPv4 wireguard interface.
   string interface_ipv4_addr = 3;
+
+  // The public key for IPv6 wireguard on this endpoint.
+  string public_key_v6 = 4;
+
+  // The IP address of the IPv6 wireguard interface.
+  string interface_ipv6_addr = 5;
 }
 
 message WireguardEndpointRemove {
   // The name of the wireguard host.
   string hostname = 1;
+
+  // The IP version of the endpoint being removed
+  IPVersion ip_version = 2;
```

Comment:
@coutinhop Since we're using the same WireguardEndpointUpdate and WireguardEndpointRemove structures for both v4 and v6, do we also need an `ip_version` field on the EndpointUpdate structure?

In other words, what are the identifiers for this struct? Hostname, or Hostname+IPVersion? 

How does the data plane use the ip_version field of the WireguardEndpointRemove message?

Reply:
My reasoning was (and you can see how the data plane would use the messages in https://github.com/projectcalico/calico/pull/6204) that for the update message, we would have either v4 or v6 data (address and public key) present, but not both. I added new fields just because I didn't think renaming interface_ipv4_address to an IP version would be a good idea (much less storing an IPv6 address in that field). But ideally it should be called simply interface_ip_address and we would infer the IP version from the value of it... Would you suggest doing things differently?

Reply:
Changed it to use proto.WireguardEndpointV6Update+proto.WireguardEndpointV6Remove messages (consistent with proto.HostMetadataV6Update/Remove messages).